pycsw/server.py

```diff
@@ -396,8 +396,16 @@ def dispatch(self, writer=sys.stdout, write_headers=True):
             rs_cls = getattr(rs_mod, rs_clsname)
 
             try:
-                self.repository = rs_cls(self.context, repo_filter)
-                LOGGER.debug('Custom repository %s loaded (%s)', rs, self.repository.dbtype)
+                connection_done=False
+                max_attempt=0
+                while not connection_done and max_attempt<=5:
```

Comment:
Should we add `max_attempt` as a configurable (i.e. in `repository.max_retries`).  Then on pycsw init, we check/set in config with a default of 3.